lib/src/components/sprite_component.dart

```diff
@@ -24,15 +24,36 @@ class SpriteComponent extends PositionComponent {
   Paint overridePaint;
 
   /// Creates a component with an empty sprite which can be set later
-  SpriteComponent();
+  SpriteComponent({
+    Vector2 position,
+    Vector2 size,
+    this.sprite,
+    this.overridePaint,
+  }) : super(position: position, size: size);
 
-  SpriteComponent.fromImage(Vector2 size, Image image)
-      : this.fromSprite(size, Sprite(image));
+  @Deprecated('Use SpriteComponent instead')
+  factory SpriteComponent.fromImage(Vector2 size, Image image) =>
+      SpriteComponent(size: size, sprite: Sprite(image));
 
+  @Deprecated('Use SpriteComponent instead')
   SpriteComponent.fromSprite(Vector2 size, this.sprite) {
     super.size.setFrom(size);
   }
 
+  factory SpriteComponent.position(
```

Comment:
because:
- better performance than `SomeComponent()..x=3 ..y=4`
- user always define `AnyComponent` with `onLoad`, and add into game like:
```
game.add(AnyComponent(x, y))
```
it's better to keep same style in library.

Reply:
trivial *.position removed

Reply:
Great, please update the PR description too :)
(and for future PRs, please don't remove the template)

Reply:
oh, forgot this. fixed.